src/app/application.cpp

```diff
@@ -146,11 +146,11 @@ Application::Application(const QString &id, int &argc, char **argv)
 
     validateCommandLineParameters();
 
-    const QString profileDir = m_commandLineArgs.portableMode
+    const QString profileDir = isPortableModeEnabled()
         ? QDir(QCoreApplication::applicationDirPath()).absoluteFilePath(DEFAULT_PORTABLE_MODE_PROFILE_DIR)
         : m_commandLineArgs.profileDir;
     Profile::initialize(profileDir, m_commandLineArgs.configurationName,
-                        m_commandLineArgs.relativeFastresumePaths || m_commandLineArgs.portableMode);
+                        m_commandLineArgs.relativeFastresumePaths || isPortableModeEnabled());
```

Comment:
you are checking the profile folder every time you call this function! you should only do that once.

Reply:
Replaced function call with `m_portableModeEnabled`:
https://github.com/Tester798/qBittorrent/blob/46b028841ebdb7f4e613f68b2b59a16cc8c0c095/src/app/application.cpp#L147